src/lesson_4/FractionalWrapper.sol

```diff
@@ -152,7 +152,15 @@ contract FractionalWrapper is ERC20 {
     ///@param owner the owning address
     ///@return shares amount of vault asset
     function withdraw(uint256 assets, address receiver, address owner) public returns (uint256 shares) {
-        shares = _convertToShares(assets);
+        shares = previewWithdraw(assets);
+
+        if (msg.sender != owner) {
+            uint256 allowed = _convertToShares(_allowance[owner][msg.sender]);
```

Comment:
The ERC4626 vault inherits from ERC20, and its tokens are denoted as `shares`. The `allowance` is then for `shares` and this conversion is not required.

Reply:
Can't recall why I would have changed that.